design/accepted/gc_plus/gc-plus-cloud.md

```diff
@@ -8,57 +8,101 @@
+
+**SLA** - 24 hours to delete a file from the moment it became eligible for deletion.
 
 ### Metrics and Logging
-- Add UGC logging and metrics to the cloud monitoring tools
-- Alerti configuration / on-call guide - for failed job runs add P3/P4 alerts 
-- Access to the cloud user metadata for debugging
+- Collect the UGC runs logs and metrics using [AWS Cloud Watch](https://docs.aws.amazon.com/lambda/latest/dg/monitoring-metrics.html)
+- Alerti configuration / on-call guide - triggering for failed job runs P3/P4 alerts (depending on SLA), instructions to the R&D on the UGC management.
+- Access to the cloud user metadata for debugging (`_lakeFS`). Where the prepare uncommitted file is saved and the run report.
 
 ### Testing
-All the tests are pre-requisites for releasing the MUGC.
+All the tests are prerequisites for releasing the MUGC.
 - Test UGC on scale (WIP)
-  - (Existing cloud user dev repo) Performance and correctness - mark only
-    - Ensure only objects that are eligible for deletion are marked for deletion
+    - (Existing cloud user dev repo) Performance and correctness - mark only
+        - Ensure only objects that are eligible for deletion are marked for deletion
 - Existing cloud user production repo (WIP)
-  - Old repository structure performance
+    - Old repository structure performance
 - Compatibility tests - UGC client and lakeFS server
-- Test sweep functionality
-- E2E tests for UGC cloud
+- Test sweep functionality on production
+- E2E tests for UGC cloud (TODO - add details)
 
 ### Performance improvements
-- In case one of the tests will suggest long running time, consider adding performance improvement:
-  1. [Optimaized listing on old repository structure](https://github.com/treeverse/lakeFS/issues/4620)
-  2. [Implement optimized run flow](https://github.com/treeverse/lakeFS/issues/4489)
-  3. [Efficient listing on committed entries](https://github.com/treeverse/lakeFS/issues/4600)
+- In case one of the tests will suggest a long running time, consider adding performance improvement:
+    1. [Optimaized listing on old repository structure](https://github.com/treeverse/lakeFS/issues/4620)
+    2. [Implement optimized run flow](https://github.com/treeverse/lakeFS/issues/4489)
+    3. [Efficient listing on committed entries](https://github.com/treeverse/lakeFS/issues/4600)
+
+From the testing UGC on a scale of existing users, it is suggested that the bottleneck on the UGC run is in listing the committed entries (meta ranges and ranges) more than 99% of the job run time. Therefore, I suggest starting with performance improvement of it.
 
 ### Milestones
 1. Pre-tests
-   * Existing cloud user tests - scale, correctness, production
-   * Performance improvements / Bug fixes (if needed)
+    * Existing cloud user tests - scale, correctness, production
+    * Performance improvements / Bug fixes (if needed)
 2. Managed UGC job infrastructure
```

Comment:
I suggest another milestone before this one: Manually weekly run of the UGC on at least 2 repos using the existing cloud infra. Once we have this, we know it's just a matter of wiring the different components together (deployment, triggering, monitoring, etc.). Currently we're still in doubt that we can even run the UGC at scale continuously. I think it's best that we first eliminate that concern, and then move to setting the cloud infrastructure..

Reply:
I'm not sure I understand what you mean. In the current tests, we run the UGC manually on the cloud. The infra for the MUGC will be different from the current one.